tools/tools.mk

```diff
@@ -0,0 +1,107 @@
+# protoc and source is used to generate code from a protobuf file and handle imports
+PROTOC_VERSION=3.14.0
+PROTOC_ZIP=protoc-$(PROTOC_VERSION)-linux-x86_64.zip
+ifeq ($(GOHOSTOS)_$(GOHOSTARCH),darwin_amd64)
+    PROTOC_ZIP=protoc-$(PROTOC_VERSION)-osx-x86_64.zip
+endif
+PROTOC_RELEASES_URI=https://github.com/protocolbuffers/protobuf/releases/download
+PROTOC_DOWNLOAD=$(PROTOC_RELEASES_URI)/v$(PROTOC_VERSION)/$(PROTOC_ZIP)
+
+tools/protoc = $(TOOLS_BIN)/protoc/${PROTOC_VERSION}/protoc
+$(tools/protoc):
+	@printf "$(OK_COLOR)==> Installing tools/protoc$(NO_COLOR)\n"
+	mkdir -p "$(@D)"
+	curl --fail --output "$@.zip" -L "$(PROTOC_DOWNLOAD)"
+	go run tools/unzip.go "$@.zip" "$@" "bin/protoc"
+	rm -f "$@.zip"
+
+PROTOBUF_GZIP=protobuf-all-$(PROTOC_VERSION).tar.gz
+PROTOBUF_DOWNLOAD=$(PROTOC_RELEASES_URI)/v$(PROTOC_VERSION)/$(PROTOBUF_GZIP)
+
+tools/protobuf-src = $(TOOLS_VENDOR)/protobuf/${PROTOC_VERSION}/protobuf.tar.gz
+$(tools/protobuf-src): $(tools/protoc)
+	@printf "$(OK_COLOR)==> Installing tools/protobuf-src$(NO_COLOR)\n"
+	mkdir -p "$(@D)"
+	rm -rf $(TOOLS_VENDOR)/protobuf/current
+	curl --fail --output "$@" -L "$(PROTOBUF_DOWNLOAD)"
+	go run tools/untargzip.go "$@" "$(@D)"
+	mv $(TOOLS_VENDOR)/protobuf/${PROTOC_VERSION}/protobuf-${PROTOC_VERSION} $(TOOLS_VENDOR)/protobuf/current
+
+tools/protobuf = $(TOOLS_VENDOR)/protobuf/current
+$(tools/protobuf): $(tools/protobuf-src)
```

Comment:
Why have this setup in this order? Would changing target on [line 22](https://github.com/hellofresh/kangal/pull/67/files#diff-c11f78e80b106456ef936228012f8d5c5aed83a8c3bf1253b359445734e451e2R22) to `$(tools/protobuf)` not make more sense since the output of line 22 is `$(tools/protobuf)`.

Reply:
The idea here is: `tools/protobuf-src` refers to a versioned dependency while `tools/protobuf` has fixed path that is prepared by versioned (we need to have fixed path to refer to it in `buf`). Versioned file keeps downloaded archive file to ensure task was already done.